resources/tests/org.eclipse.core.tests.resources/src/org/eclipse/core/tests/resources/regression/Bug_297635.java

```diff
@@ -0,0 +1,155 @@
+/*******************************************************************************
+ *  Copyright (c) 2010, 2015 IBM Corporation and others.
+ *
+ *  This program and the accompanying materials
+ *  are made available under the terms of the Eclipse Public License 2.0
+ *  which accompanies this distribution, and is available at
+ *  https://www.eclipse.org/legal/epl-2.0/
+ *
+ *  SPDX-License-Identifier: EPL-2.0
+ *
+ *  Contributors:
+ *     IBM Corporation - initial API and implementation
+ *     Alexander Kurtakov <akurtako@redhat.com> - Bug 459343
+ *******************************************************************************/
+package org.eclipse.core.tests.resources.regression;
+
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import java.lang.reflect.Field;
+import java.util.function.Consumer;
+import org.eclipse.core.internal.resources.SaveManager;
+import org.eclipse.core.resources.ISaveContext;
+import org.eclipse.core.resources.ISaveParticipant;
+import org.eclipse.core.resources.IWorkspace;
+import org.eclipse.core.runtime.CoreException;
+import org.eclipse.core.runtime.Platform;
+import org.eclipse.core.tests.harness.BundleTestingHelper;
+import org.eclipse.core.tests.resources.ResourceTest;
+import org.eclipse.core.tests.resources.content.ContentTypeTest;
+import org.osgi.framework.Bundle;
+import org.osgi.framework.BundleContext;
+import org.osgi.framework.BundleException;
+
+/**
+ * Tests regression of bug 297635
+ */
+public class Bug_297635 extends ResourceTest {
+
+	@Override
+	protected void setUp() throws Exception {
+		super.setUp();
+		BundleWithSaveParticipant.install();
+		saveFull();
+	}
+
+	@Override
+	protected void tearDown() throws Exception {
+		BundleWithSaveParticipant.uninstall();
+		super.tearDown();
```

Comment:
In JUnit5 (if we ever migrate to it), an Exception in the `setup` method could leave the field `bundle` empty and calling `uninstall` would throw an `IllegalStateException` and would keep the test from cleaning up properly (no call to `super.tearDown()` would happen). 

Is this intentional? If not, I would just invert the order of these 2 lines.

Reply:
Thanks for the finding! That was not intentional but a "regression" I introduced with the recent change. I fixed that in 19dbe22.